File: SecondYear/Anantj2499_AnantJaiswal_2226cseml1088_2/Week_2/CheckArmstrong.java

    import java.util.*;
    public class CheckArmstrong {

        public static void main(String[] args) {
            // Write your code here
            Scanner sc = new Scanner(System.in);
            int n = sc.nextInt();
            int temp = n;
            int count = 0;
            while (temp > 0) {
                temp /= 10;
                count++;
            }
            temp = n;
            int sum = 0;
            int digit;
            while (temp > 0) {
                digit = temp % 10;
                sum += (int) Math.pow(digit, count);
                temp /= 10;
            }
            if (sum == n)
                System.out.println("true");
            else
                System.out.println("false");
        }
    }

File: SecondYear/Junior-Hawk_ArchitAgarwal_2226cseai1186_2/Week_2/check_armstrong.cpp
int countDigits(int num) {
    int count = 0;
    while (num > 0) {
        num /= 10;
        count++;
    }
    return count;
}
bool checkArmstrong(int n){
	int originalNum = n;
    int c = countDigits(n);
    int sum = 0;
    
    while (n > 0) {
        int digit = n % 10;
        sum += pow(digit, c);
        n /= 10;
    }
    
    return sum == originalNum;
}
